lib/ipam/ipam.go

```diff
@@ -603,26 +646,31 @@ func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, a
 	}
 
 	// Allocate the IPs.
-	for len(ips) < num {
+	for len(ia.IPs) < num {
 		var b *model.KVPair
 
-		rem := num - len(ips)
+		rem := num - len(ia.IPs)
 		if maxNumBlocks > 0 && numBlocksOwned >= maxNumBlocks {
 			s.allowNewClaim = false
 		}
 
 		b, newlyClaimed, err := s.findOrClaimBlock(ctx, 1)
 		if err != nil {
 			if _, ok := err.(noFreeBlocksError); ok {
+				if config.StrictAffinity {
+					ia.AddMsg("No more free affine blocks and strict affinity enabled")
```

Comment:
Should we also add a message for the case where it's not enforcing `StrictAffinity` which should mean we couldn't find any unclaimed blocks?

Reply:
We would break out of this loop in either case, but we would check non-affine blocks further below (if strict affinity is not enabled). Do you think we should change this to "No more free affine blocks, strict affinity: %v" and add this message independent of strict affinity?

Reply:
oh i think I misread `findOrClaimBlock` yeah I don't think my original comment matters here.